src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/authorization/ReferralAccessChecker.kt

```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.authorization
+
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+
+class ReferralAccessChecker(
```

Comment:
🤔 I'm wondering if this has to be an application layer concern.

Does it make sense to say: we only get back objects from the database that are valid for the user's validated scope?

I.e. should we have `findById(id)` and `findByIdAndScope(id, userScope)`; and similar for the list lookups? We then could construct a query that effectively implements `eligibleProviderMapper` as filter conditions.

The net benefit of this approach could be: application code could rely that it always gets properly scoped data, unless I'm wrong in my reasoning here

Reply:
i'd be happy to go down this route. it needs more thinking though. i don't think we should be blocked on getting some kind of implementation rolled out in the short term.

e.g. with this idea, how do we differentiate between referral ids that don't exist and access being restricted in the application layer?

Reply:
after a lengthy discussion about this we determined that utilizing the components here will well as long as the implementation is 'hidden' in the service layer. 

i.e. controllers still call 'user scoped' service layer methods like "get sent referral for user", which handle all this complexity.